paddle/fluid/imperative/gradient_accumulator.cc

```diff
@@ -100,6 +107,51 @@ class TensorAddFunctor : public boost::static_visitor<> {
   T* y_;
 };
 
+template <typename DeviceContext, typename T>
+void TensorAddImpl(const framework::Tensor& src, framework::Tensor* dst,
+                   const platform::Place& place) {
+  platform::DeviceContextPool& pool = platform::DeviceContextPool::Instance();
+  paddle::platform::DeviceContext* ctx = pool.Get(place);
+  auto dev_ctx = dynamic_cast<DeviceContext*>(ctx);
+  switch (src.dims().size()) {
+    case 1: {
+      operators::math::AddTensor<DeviceContext, T, 1> func;
+      func(dev_ctx, src, dst);
+      break;
+    }
+    case 2: {
+      operators::math::AddTensor<DeviceContext, T, 2> func;
+      func(dev_ctx, src, dst);
+      break;
+    }
+    case 3: {
+      operators::math::AddTensor<DeviceContext, T, 3> func;
+      func(dev_ctx, src, dst);
+      break;
+    }
+    case 4: {
+      operators::math::AddTensor<DeviceContext, T, 4> func;
+      func(dev_ctx, src, dst);
+      break;
+    }
+    case 5: {
+      operators::math::AddTensor<DeviceContext, T, 5> func;
+      func(dev_ctx, src, dst);
+      break;
+    }
+    case 6: {
+      operators::math::AddTensor<DeviceContext, T, 6> func;
+      func(dev_ctx, src, dst);
+      break;
+    }
+    default:
+      PADDLE_THROW(platform::errors::InvalidArgument(
+          "TensorAddImpl doesn't supports tensors whose ranks are greater "
+          "than 6."));
```

Comment:
why doesn't support?

Reply:
Not sure, there are several operators which check `rank <= 6`, for example, `transpose` op. I will remove `switch...case...` and simplify the code.